server/sonar-server/src/main/java/org/sonar/server/metric/ws/SearchAction.java

```diff
@@ -83,7 +82,7 @@ public void handle(Request request, Response response) throws Exception {
     Boolean isCustom = request.paramAsBoolean(PARAM_IS_CUSTOM);
     DbSession dbSession = dbClient.openSession(false);
     try {
-      List<MetricDto> metrics = dbClient.metricDao().selectEnabled(dbSession, isCustom, searchOptions.getOffset(), searchOptions.getLimit());
+      List<MetricDto> metrics = dbClient.metricDao().selectEnabledSortedByKey(dbSession, isCustom, searchOptions.getOffset(), searchOptions.getLimit());
```

Comment:
I know that we've talked about this, but in fact you should keep the sort by name from DB : every metrics should have a name in DB, but they have not necessarily a name in the bundle.
Let's talk about that.